### PR TITLE
add nicknames & pfps!

### PR DESCRIPTION
**Pull request by bitful-pannul**
See https://github.com/bitful-pannul/nimi/ README for details :D

### CODE REVIEW

**Comment by bitful-pannul** [comment]
note, might want to update zig lib and sur files to 413 versions, but they work right now too.

File: con/minter.hoon [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    =/  =item
+      :*  %&  id
+          this.context
+          nft.act
+          town.context
+          name.act
+          %lookup  [id.caller.context next-item ship.act]
+      ==  
+    =/  propmap  %-  make-pmap
+      ~[[%name name.act]]
```
Comment (dr-frmr) [inline con/minter.hoon:44]: I don't see why this can't just be put in the one spot it gets used

File: con/lib/minter.hoon [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          name=@t
+          uri=@t
+          ship=(unit @p)
+      ==
+      ::
+      $:  %on-push
+          from=id
+          amount=@ud
+          calldata=*
+  ==  ==
```
Comment (dr-frmr) [inline con/lib/minter.hoon:15]: these two action defs can probably both be one-line each, I only use the tall-form when there are a ton of arguments

File: app/nimi.hoon [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+      |=  =ship
+      :*  %pass   /disme
+          %agent  [ship %nimi]
+          %poke   %nimi-action
+          !>([%whodis ship])
+      ==
+    ::
+    :_  state
+    pokes
+    ::
```
Comment (dr-frmr) [inline app/nimi.hoon:192]: the `::` lines that go *between* pokes should be on the same indent-level as the `?-`, much clearer

File: app/nimi.hoon [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+    ?~  up  ``nimi-update+!>(`update`[%no-user ~])
+    ?>  ?=(%newest-item -.up)
+    =+  item=item.up
+    ?>  ?=(%.y -.item)
+    ::
+    =/  data  ;;([address=@ux item=@ux ship=(unit @p)] noun.p.item)
+    ::  fix
+    =/  user  [address.data item.data ship.data ~ ~]
+    ::  
+    ::
```
Comment (dr-frmr) [inline app/nimi.hoon:382]: no double-empty-comment lines!

File: app/nimi.hoon [hunk 5]
```diff
@@ -0,0 +1,437 @@
...
+      |=  =ship
+      :*  %pass   /disme
+          %agent  [ship %nimi]
+          %poke   %nimi-action
+          !>([%disme [0x0 0x0 [0 0 0]]])
+      ==
+    ::
+    =/  i  (name-from-item item.act `address.act)  :: destructure
+    ::
+    =/  name  (fall name.i '')
```
Comment (dr-frmr) [inline app/nimi.hoon:102]: probably too verbose

File: app/nimi.hoon [hunk 6]
```diff
@@ -0,0 +1,437 @@
...
+    ?:  ?&  ?=(^ user)
+            =(address.u.user address.act)
+            =(item.u.user item.act)
+        ==
+      `state
+    ::
+    =/  [cards=(list card) =profile]  (validate-scry +.act)
+    [cards state(niccbook (~(put by niccbook) src.bowl profile))]
+    ::
+    ::
```
Comment (dr-frmr) [inline app/nimi.hoon:131]: x

**Review by dr-frmr — APPROVED** [review]
LFG

**Pull request merged**